port/netpacket.c

```diff
@@ -242,15 +242,27 @@ void netpacket_linkoutput(struct netif *netif, struct pbuf *p)
 		SMEMCPY(&src_addr, netif->hwaddr, sizeof(struct eth_addr));
 
 		if (pcb->type == SOCK_DGRAM) {
-			/* remove ethernet header from pbuf */
-			pbuf_header(p, -(s16_t)sizeof(struct eth_hdr));
+			/* remove ethernet header */
+			pbuf_remove_header(p, sizeof(struct eth_hdr));
```

Comment:
Just noticed... Why there was "-(s16_t)sizeof(struct eth_hdr)"? This is substantial change (now it's straight sizeof), is this ok?

Reply:
It OK, pbuf_header(p, -x) calls pbuf_remove_header(p, x). I just wanted to make it clear we are removing the header here.

Reply:
Oh, ok. What a weird API